Implement bandwidth monitoring.

clalancette opened the pull request:
We now have all of the functionality available to do this,
so finish implementing it.  Note that in order to do this,
I changed the TopicInfo class from a "is-a" ROSTopicHz to
a "has-a" ROSTopicHz, and also added a "has-a" ROSTopicBandwidth.
With all of that in place, it is pretty simple to fetch
the bandwidth data and display it.

Signed-off-by: Chris Lalancette <clalancette@openrobotics.org>

Note that this requires https://github.com/ros2/ros2cli/pull/708 , so if we want to backport this to Humble we also need to backport that PR.

ivanpauno reviewed (changes requested):
Code LGTM!

> so if we want to backport this to Humble we also need to backport that PR.

Note that this PR is targeting `foxy-devel`, so we need either to branch off or to backport the ros2cli changes also to `foxy` and `galactic`.
Branching off seems like the safest thing to do.

clalancette commented:
> Note that this PR is targeting `foxy-devel`, so we need either to branch off or to backport the ros2cli changes also to `foxy` and `galactic`.
> Branching off seems like the safest thing to do.

Oof, yes, great point.  We should branch off.

ivanpauno commented:
I see there's already a [humble](https://github.com/ros-visualization/rqt_topic/tree/humble) branch, so we need to retarget this PR there and backport the ros2cli change to humble, or create a different branch here for rolling.

clalancette commented:
CI:
* Linux [![Build Status](http://ci.ros2.org/buildStatus/icon?job=ci_linux&build=16893)](http://ci.ros2.org/job/ci_linux/16893/)
* Linux-aarch64 [![Build Status](http://ci.ros2.org/buildStatus/icon?job=ci_linux-aarch64&build=11463)](http://ci.ros2.org/job/ci_linux-aarch64/11463/)
* Windows [![Build Status](http://ci.ros2.org/buildStatus/icon?job=ci_windows&build=17296)](http://ci.ros2.org/job/ci_windows/17296/)

ivanpauno reviewed (approved):
LGTM!

clalancette commented:
Silly flake8 error.  Rebased, and fixed.  Here is another CI:

* Linux [![Build Status](http://ci.ros2.org/buildStatus/icon?job=ci_linux&build=16896)](http://ci.ros2.org/job/ci_linux/16896/)
* Linux-aarch64 [![Build Status](http://ci.ros2.org/buildStatus/icon?job=ci_linux-aarch64&build=11466)](http://ci.ros2.org/job/ci_linux-aarch64/11466/)
* Windows [![Build Status](http://ci.ros2.org/buildStatus/icon?job=ci_windows&build=17299)](http://ci.ros2.org/job/ci_windows/17299/)
